### PR TITLE
sys/include/xtimer.h: deprecate nanosleep

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR deprecates `nanosleep`, realistically this made no sense since no backend was able at running at this speed.

### Testing procedure

A green ci should be enough...

### CODE REVIEW

File: sys/include/xtimer.h [hunk 1]
```diff
@@ -176,6 +176,10 @@ static inline void xtimer_usleep64(uint64_t microseconds);
 /**
  * @brief Stop execution of a thread for some time
  *
+ * @deprecated This function is deprecated as now XTIMER backend is currently
+ *             configured to ron at more than 1Mhz, making nanoseconds accuracy
+ *             impossible to achieve.
+ *
```
Comment (basilfx) [inline sys/include/xtimer.h:182]:
```suggestion
 * @deprecated This function is deprecated as no XTIMER backend is currently
 *             configured to run at more than 1 MHz, making nanoseconds accuracy
 *             impossible to achieve.
 *
```

**Review by maribu — APPROVED** [review]
ACK. Please squash in the typo fixes right away.

File: drivers/soft_spi/Kconfig [hunk 2]
```diff
@@ -11,3 +11,7 @@ config MODULE_SOFT_SPI
     depends on TEST_KCONFIG
     select MODULE_PERIPH_GPIO
     select MODULE_XTIMER
+
+config MODULE_SOFT_SPI_AS_PERIPH_SPI
+    bool "Software-implemented SPI implements periph/spi API"
+    select MODULE_SOFT_SPI
```
Comment (benpicco) [inline drivers/soft_spi/Kconfig:17]: Where did that come from? :wink:

Comment (fjmolinas) [reply drivers/soft_spi/Kconfig]: Ups was looking at @maribu soft-spi PR at the same time...removed

**Pull request merged**